src/modules/tools/endstops/Endstops.cpp

```diff
@@ -1098,33 +1098,34 @@ void Endstops::on_gcode_received(void *argument)
                 gcode->stream->printf(" will take effect next home\n");
                 break;
 
-            case 211: { //Enable or disable endstop(s) hard limit
-                if(gcode->has_letter('S')) { //Alter endstop behavior    
-                        for(auto& e : endstops) {
-                            if(gcode->has_letter(e->axis)) {
-                                if(e->is_max_stop == gcode->get_value(e->axis)) {  //Are we configuring a min or max endstop?
-                                    string name;
-                                    name.append(1, e->axis).append(e->is_max_stop ? "_max" : "_min");
-                                    gcode->stream->printf("%s: %s ", name.c_str(), (e->limit_enable?"enabled":"disabled"));
-                                    THECONVEYOR->wait_for_idle();
-                                    e->limit_enable = gcode->get_value('S')>0;
-                                    if(!limits_enabled && e->limit_enable) {
-                                        register_for_event(ON_IDLE);
-                                        limits_enabled = true;
+            case 211: //Enable or disable endstop(s) hard limit
+                if(gcode->subcode == 1) { //M211.1 for hardware endstop manipulation
+                    if(gcode->has_letter('S')) { //Alter endstop behavior
+                            for(auto& e : endstops) {
+                                if(gcode->has_letter(e->axis)) {
+                                    if(e->is_max_stop == (gcode->get_value(e->axis)>0)) {  //Are we configuring a min or max endstop?
```

Comment:
I still do not like this, the code is obfuscated and not clear what you are doing. Also it is still not safe to compare a float with 0. To do that safely you need to use gcode->get_uint(..).
I had to read this several times to understand what it is supposed to do, and as such it means others will not have a clue! Sometimes it is better to be verbose rather than do some neat coding trick, especially when others have to read the code.

Reply:
Understood, will verbosify and uint it.